.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,210 @@
+            echo "latest-tag=$tag" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Build Changelog
+        id: build-changelog
+        env:
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+        run: |
+          CHANGELOG=$(curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ github.repository }}/releases/generate-notes \
+            -d '{"tag_name":"${{ env.RELEASE_VERSION }}","target_commitish":"${{ env.RELEASE_BRANCH }}","previous_tag_name":"${{ env.PREVIOUS_VERSION }}","configuration_file_path":".github/release.yml"}' \
+            | jq -r '.body')
+
+          # The EOF steps are used to save multiline string in github:
+          # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "changelog<<$EOF" >> $GITHUB_OUTPUT
+          echo -e "${CHANGELOG}" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
+
+      - name: Update Changelog
+        if: ${{ env.PRE_RELEASE_VERSION == '' }}
+        env:
+          CHANGELOG_CONTENT: ${{ steps.build-changelog.outputs.changelog }}
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+        run: |
+          echo -e "# ${RELEASE_VERSION}\n\n${CHANGELOG_CONTENT}" | cat - CHANGELOG.md > temp && mv temp CHANGELOG.md
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Update Version References in Source
+        env:
+          RELEASE_TAG: ${{ steps.set-release.outputs.release-tag }}
+        run: |
+          rm -rf dist node_modules
+          npm install
+          npm run build
+          NEW_HASH=$(cat dist/browser/algosdk.min.js | openssl dgst -sha384 -binary | openssl base64 -A)
+          python3 scripts/bump_version.py ${RELEASE_TAG:1} --new_hash "${NEW_HASH}"
+          npx prettier --write .
+
+      - name: Commit Changes
+        uses: EndBug/add-and-commit@v9.1.3
+        env:
+          RELEASE_TAG: ${{ steps.set-release.outputs.release-tag }}
+        with:
+          message: "bump up version to ${{ env.RELEASE_TAG }}"
+
+      - name: Create Pull Request to Master
+        env:
+          CHANGELOG_CONTENT: ${{ steps.build-changelog.outputs.changelog }}
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+          GH_TOKEN: ${{ github.token }}
+          RELEASE_TAG: ${{ steps.set-release.outputs.release-tag }}
+        run: |
+          # Note: There's an issue adding teams as reviewers, see https://github.com/cli/cli/issues/6395
+          PULL_REQUEST_URL=$(gh pr create --base "master" \
+            --title "FOR REVIEW ONLY: ${{ github.event.repository.name }} $RELEASE_TAG" \
```

Comment:
This is something I've never understood: why do we label the release PRs with "FOR REVIEW ONLY" when we clearly intend them to be merged like normal PRs? Is this a relic from an older release process?

Reply:
Hmmm... this is a good point.
I know this is related to the "git flow" process we're using and how we tag after we merge the release branch into master. I wonder if preventing the merge via UI is because of the "git flow" process or the `git-flow` command line tool we were using which merged and tags the branches.
We are not using the `gitflow` cli in the new automation, so maybe it's not necessary to prevent it from merging...

I'm going to have to get back to you on this.

Reply:
So it seems to be related to the branch restrictions:
- Previously we were using admin privileges to merge in develop and override the "linear history" requirement, but I think from now on, we'll try to stick to the "linear history" restriction on the "develop" branch which requires a rebase if there were any new changes in develop before we merge back.
- For "master" branch, I think it's to make sure we don't accidentally use the default "Squash and Merge" method since we want the individual commits in the branch.

Reply:
Thanks for the background.

Can you clarify whether the develop PR created by this workflow will actually be merged or not? And are you suggesting rebasing develop off of master if it has additional commits? Changing history seems very undesirable for our most active branch.

Reply:
Yes develop PR will be merged. We would rebase the release branch before merging it into develop if there were new changes on `develop` after the release branch was initially cut.

If there were no changes on develop, a rebase is not necessary and it should merge back fine. Although [according to this post](https://stackoverflow.com/questions/60597400/how-to-do-a-fast-forward-merge-on-github), it may just rebase back as well. But we also don't want anyone to "Squash and Merge" for the merge back to keep the original commits that went into master branch.